### ISSUE TITLE
Issue with decode in test_review when deploying the model using PyTorch?

### ISSUE THREAD

**Issue by hpkhanh1610** [udacity/sagemaker-deployment#1]
Hi, 

In the project, at Step 6 (again) - Deploy the model for the web app, at the function **def test_reviews(data_dir='../data/aclImdb', stop=250):**, 
`` # Read in the review and convert to 'utf-8' for transmission via HTTP
                review_input = review.read().encode('utf-8')
                # Send the review to the predictor and store the results
                results.append(int(predictor.predict(review_input))) ``

I think the last line should be:
**results.append(round(float(predictor.predict(review_input))))**

The output from predictor.predict is in byte type, that is **b' '**, so we need to use **float()** to convert to real number and use **round** to round it to the nearest integer (**int()** is not correct in this case because it rounds all numbers from 0-1 down to 0 only).

**Comment by preetkhaturia** [comment]
Did you solve this? I am having the same issue.

**Comment by preetkhaturia** [comment]
Can we see the intermediate outputs in predict function?? Because I am not really sure, I am having an error due to rounding.

**Comment by hpkhanh1610** [comment]
The intermediate part is similar to what we need to do in the notebook in **(TODO) More testing**  before we get to the predict.py:

input_data_cleaned = review_to_words(input_data)
data_X, data_len = convert_and_pad(model.word_dict, input_data_cleaned)
...
result = model(data).cpu().detach()
result = np.array(round(result.item()))
